src/Tasks/Microsoft.NET.Build.Tasks/CacheArtifactParser.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+using NuGet.Versioning;
+using NuGet.Packaging.Core;
+
+#if PRODUCT
```

Comment:
I disagree.  This is putting extra constraints on our product code that don't need to be there.  For example these #ifs are unnecessary.  If we want to add more logic to this file, we will have to continue adding these #ifs.

I still think baselining a couple xml files is the correct way for the tests to verify the output xml file is correct.  It makes it super easy to see what has changed in the file when it has changed.  And it catches things like when unexpected elements start showing up in the file.

---
In reply to: [103600599](https://github.com/dotnet/sdk/pull/890#discussion_r103600599) [](ancestors = 103600599)